### PR TITLE
Enable Dynamic Camera

### PR DESCRIPTION
**Pull request by tchabot22**
Added StatefulWidget to GoogleAtlas to track position and provide map position updates when camera position is changed.  
Added FakeMapController to assist with testing.  
Transitioned GoogleAtlas unit tests to widget tests to better evaluate functionality.

### CODE REVIEW

File: packages/google_atlas/test/widget_tests/google_atlas_widget_test.dart [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+
+    testWidgets(
+        'should return correct GoogleMap when build is called with onTap',
+        (WidgetTester tester) async {
+      GoogleMaps.CameraPosition expectedCameraPosition =
+          GoogleMaps.CameraPosition(
+        target: GoogleMaps.LatLng(41.8781, -87.6298),
+        zoom: 14.4746,
+      );
+      /* int onTapCallCount = 0;
```
Comment (jorgecoca) [inline packages/google_atlas/test/widget_tests/google_atlas_widget_test.dart]: Remove commented code

**Review by jorgecoca — CHANGES_REQUESTED** [review]

File: packages/google_atlas/lib/src/google_atlas.dart [hunk 2]
```diff
@@ -12,13 +12,64 @@ class GoogleAtlas extends Provider {
...
+      zoom: 10,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _onPositionUpdate();
+
     return GoogleMaps.GoogleMap(
+      key: Key("GoogleMap"),
```
Comment (felangel) [inline packages/google_atlas/lib/src/google_atlas.dart]: do we still need this key? I'd prefer it be dynamic as opposed to hard-coded

File: packages/google_atlas/lib/src/google_atlas.dart [hunk 3]
```diff
@@ -12,13 +12,64 @@ class GoogleAtlas extends Provider {
...
       myLocationEnabled: showMyLocation,
       myLocationButtonEnabled: showMyLocationButton,
       mapType: GoogleMaps.MapType.normal,
       initialCameraPosition: _toGoogleCameraPosition(cameraPosition),
-      markers: markers.map((m) => _toGoogleMarker(m)).toSet(),
+      markers: this.markers.map((m) => _toGoogleMarker(m)).toSet(),
```
Comment (felangel) [inline packages/google_atlas/lib/src/google_atlas.dart]: we don't need `this`

File: packages/google_atlas/lib/src/google_atlas.dart [hunk 4]
```diff
@@ -12,13 +12,64 @@ class GoogleAtlas extends Provider {
...
+}
+
+class _GoogleMapsProviderState extends State<GoogleMapsProvider> {
+  CameraPosition get cameraPosition => widget.cameraPosition;
+  CameraPosition currentPosition;
+  Set<Marker> get markers => widget.markers;
+  bool get showMyLocation => widget.showMyLocation;
+  bool get showMyLocationButton => widget.showMyLocationButton;
+  ArgumentCallback<LatLng> get onTap => widget.onTap;
+  GoogleMaps.GoogleMapController mapController;
```
Comment (felangel) [inline packages/google_atlas/lib/src/google_atlas.dart]: can we make `mapController` and `currentPosition` private?

File: packages/google_atlas/test/widget_tests/google_atlas_widget_test.dart [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+
+      final FakePlatformGoogleMap platformGoogleMap =
+          fakePlatformViewsController.lastCreatedView;
+
+      expect(platformGoogleMap.cameraPosition, expectedCameraPosition);
+    });
+  });
+}
+
+/// Atlas Test Sample Widget
```
Comment (felangel) [inline packages/google_atlas/test/widget_tests/google_atlas_widget_test.dart]: do we need this comment?

**Review by felangel — CHANGES_REQUESTED** [review]
see comments

**Review by jorgecoca — APPROVED** [review]

**Review by felangel — APPROVED** [review]
lgtm

**Comment by felangel** [comment]
did you want to update the pubspec version and changelog as well?

**Pull request merged**